qcfractal/tests/test_collections.py

```diff
@@ -178,3 +196,20 @@ def test_compute_openffworkflow(fractal_compute_server):
     assert final_energies.keys() == {butane_id, "HOOH"}
     assert final_energies[butane_id].keys() == {"label1"}
     assert final_energies[butane_id]["label1"] is None
+
+    optimization_input = {
+        "label2": {
+            "type": "optimization_input",
+            "initial_molecule": butane.to_json(),
+            "constraints": {'scan': [('dihedral', '0', '2', '3', '1', '0', '120', '4')]}
+        }
+    }
+
+    wf.add_fragment(butane_id, optimization_input, provenance={})
+    fractal_compute_server.await_services(max_iter=5)
+
+    final_energies = wf.list_final_energies()
```

Comment:
A bit surprised this works, what does final_energies show for the `optimization`? You will likely have to go into `list_final_energies` and have it modify based on torsiondrive/optimization.